service/systemd/service_test.go

```diff
@@ -251,445 +197,425 @@ exec 2>&1
-
+	svc := s.newService(c)
+	svc.Service.Conf = common.Conf{}
+	_, err := svc.Exists()
 	c.Check(err, gc.ErrorMatches, `.*no conf expected.*`)
-	s.stub.CheckCalls(c, nil)
 }
 
 func (s *initSystemSuite) TestRunningTrue(c *gc.C) {
-	s.addService("jujud-machine-0", "active")
-	s.addService("something-else", "error")
-	s.addService("juju-mongod", "active")
+	ctrl := s.patch(c)
+	defer ctrl.Finish()
+
+	gomock.InOrder(
+		s.dBus.EXPECT().ListUnits().Return([]dbus.UnitStatus{
+			{Name: "jujud-machine-0.service", LoadState: "loaded", ActiveState: "active"},
+			{Name: "juju-mongod.service", LoadState: "loaded", ActiveState: "active"},
+		}, nil),
+		s.dBus.EXPECT().Close(),
+	)
 
-	running, err := s.service.Running()
+	running, err := s.newService(c).Running()
 	c.Assert(err, jc.ErrorIsNil)
-
 	c.Check(running, jc.IsTrue)
-	s.stub.CheckCallNames(c, "ListUnits", "Close")
 }
 
 func (s *initSystemSuite) TestRunningFalse(c *gc.C) {
-	s.addService("jujud-machine-0", "inactive")
-	s.addService("something-else", "error")
-	s.addService("juju-mongod", "active")
+	ctrl := s.patch(c)
+	defer ctrl.Finish()
+
+	gomock.InOrder(
+		s.dBus.EXPECT().ListUnits().Return([]dbus.UnitStatus{
+			{Name: "jujud-machine-0.service", LoadState: "loaded", ActiveState: "inactive"},
+			{Name: "juju-mongod.service", LoadState: "loaded", ActiveState: "active"},
+		}, nil),
+		s.dBus.EXPECT().Close(),
+	)
 
-	running, err := s.service.Running()
+	running, err := s.newService(c).Running()
 	c.Assert(err, jc.ErrorIsNil)
-
 	c.Check(running, jc.IsFalse)
-	s.stub.CheckCallNames(c, "ListUnits", "Close")
 }
 
 func (s *initSystemSuite) TestRunningNotEnabled(c *gc.C) {
-	s.addService("something-else", "active")
+	ctrl := s.patch(c)
+	defer ctrl.Finish()
+
+	gomock.InOrder(
+		s.dBus.EXPECT().ListUnits().Return([]dbus.UnitStatus{
+			{Name: "random-thing.service", LoadState: "loaded", ActiveState: "active"},
+		}, nil),
+		s.dBus.EXPECT().Close(),
+	)
 
-	running, err := s.service.Running()
+	running, err := s.newService(c).Running()
 	c.Assert(err, jc.ErrorIsNil)
-
 	c.Check(running, jc.IsFalse)
-	s.stub.CheckCallNames(c, "ListUnits", "Close")
 }
 
 func (s *initSystemSuite) TestRunningError(c *gc.C) {
-	s.addService("jujud-machine-0", "active")
-	s.addService("something-else", "error")
-	s.addService("juju-mongod", "active")
-	failure := errors.New("<failed>")
-	s.stub.SetErrors(failure)
+	ctrl := s.patch(c)
+	defer ctrl.Finish()
 
-	running, err := s.service.Running()
-	c.Assert(errors.Cause(err), gc.Equals, failure)
+	gomock.InOrder(
+		s.dBus.EXPECT().ListUnits().Return(nil, errFailure),
+		s.dBus.EXPECT().Close(),
+	)
 
-	c.Check(running, jc.IsFalse)
-	s.stub.CheckCallNames(c, "ListUnits", "Close")
+	_, err := s.newService(c).Running()
+	c.Check(errors.Cause(err), gc.Equals, errFailure)
 }
 
 func (s *initSystemSuite) TestStart(c *gc.C) {
-	s.addService("jujud-machine-0", "inactive")
-	s.ch <- "done"
-	s.addListResponse()
+	ctrl := s.patch(c)
+	defer ctrl.Finish()
```

Comment:
Does the fishish function check the expected function calls defined below?

Reply:
Yes, and also fails the test if calls were received that were not expected.